validation/src/lib.rs

```diff
@@ -193,6 +193,8 @@ pub trait Network {
 pub struct GroupInfo {
 	/// Authorities meant to check validity of candidates.
 	pub validity_guarantors: HashSet<SessionKey>,
+	/// Mapping from validator index to public key.
+	pub index_mapping: HashMap<usize, SessionKey>,
```

Comment:
I'm not sure about this: probably `GroupInfo` fields should be made private?

Reply:
yeah, probably safer to make it private.